licenses/check_licenses.dart

```diff
@@ -124,7 +124,7 @@ Future<void> verifyNoMissingLicense(String workingDirectory, {bool checkMinimums
     workingDirectory,
     'sh',
     overrideMinimumMatches ?? 1,
-    '#!/bin/bash\n' + _generateLicense('# '),
+    '#!.*sh\n' + _generateLicense('# '),
```

Comment:
Can we use the full path for both to make sure an exact match?  Or instead, have the `build.sh` to follow the old format, and then create a followup PR to update all .sh to use the new path? Which seems the way from the framework repo.

Reply:
Okay, although it seems like it could even just be `.*\n`, since it seems odd to me that a script for checking licenses is enforcing which shells are allowed in scripts.

Reply:
How about if I just update all of the existing scripts to use `#!/usr/bin/env bash`, but leave this as-is? I don't think we should be limiting the allowed shells, or at least we shouldn't be doing it here.

Reply:
(And I'll move it into another PR.)

Reply:
Updating to use `#!/usr/bin/env bash` sounds good to me. But for the license check, can we follow what framework is doing: https://github.com/flutter/flutter/blob/master/dev/bots/analyze.dart#L642

I am not sure about the guidance here. maybe @Hixie ?

Reply:
Done, but I do think that if we want to restrict the shells used, we should be doing it in the analyze script, not in the license check.

Reply:
I reverted all the shebang-line changes, and moved them to another PR: https://github.com/flutter/cocoon/pull/2974